src/main/java/io/reactivex/Optional.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivex;
+
+import io.reactivex.internal.functions.Objects;
+
+/**
+ * Simplified backport of Java 8's Optional type.
+ *
+ * @param <T> the value type
+ */
+public final class Optional<T> {
+    final T value;
+    protected Optional(T value) {
+        this.value = value;
+    }
+    
+    static final Optional<Object> EMPTY = new Optional<Object>(null);
+    
+    @SuppressWarnings("unchecked")
+    public static <T> Optional<T> empty() {
+        return (Optional<T>)EMPTY;
+    }
+    
+    public static <T> Optional<T> of(T value) {
+        Objects.requireNonNull(value, "value is null");
+        return new Optional<T>(value);
+    }
+    
+    public boolean isPresent() {
+        return value != null;
+    }
+    
+    public T get() {
+        return value;
```

Comment:
Yup, exactly, deserves a `throw` in my opinion, to prevent leaking that `null` into user's code. 

Also 
![1454550686518](https://cloud.githubusercontent.com/assets/967132/12830813/453683d8-cba2-11e5-97f8-b820117907e1.jpg)